### PR TITLE
show a loader when compositions are having a hard time loading

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- fixes bug where we'd get an error message "not found" when changing composition name
- shows a loader when composition turns to undefined
- shows "not found" error after timeout (15 seconds!) or when component starts undefined.

### CODE REVIEW

**Comment by KutnerUri** [comment]
@GiladShoham - I'm getting an error on testing-library expect:
```tsx
scopes/react/ui/loader-fallback/loader-fallback.spec.tsx:13:20 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'Matchers<HTMLElement>'.

13     expect(result).toBeInTheDocument();
                      ~~~~~~~~~~~~~~~~~
```

didn't you fix this by adding `@testing-library/jest-dom` directly to the component dependencies?

**Comment by jonnysas** [comment]
Task linked: [CU-mrudz7 when editting a composition's function name, I get a flash of the 404 page before I get the changes](https://app.clickup.com/t/mrudz7)

File: scopes/react/ui/loader-fallback/loader-fallback.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Renders a component, and show a placeholder when it is undefined.
+
+Logic is as follows:
+
+1. when component is defined -> render it
+1. when the initial value of the component is undefined -> show the default immediately
+1. when the component had just changed to be undefined -> show Loader for _x_ seconds
+   1. then, after _x_ seconds - show NotFound
+
```
Comment (odedre) [inline scopes/react/ui/loader-fallback/loader-fallback.docs.mdx:11]: why all `1.`'s?

File: scopes/react/ui/loader-fallback/loader-fallback.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      const tmId = setTimeout(() => setRenderTarget(<DefaultComponent />), timeout);
+      return () => clearTimeout(tmId);
+    }
+    // else -> Target === undefined, prev === undefined
+
+    // comp changed from undefined to undefined
+    // could happen on first render, or when one of the other values change.
+
+    return () => {}; // nothing to do here
+  }, [Target, DefaultComponent, Loader, timeout]);
```
Comment (odedre) [inline scopes/react/ui/loader-fallback/loader-fallback.tsx:44]: why not just `Target`?

**Review by odedre — APPROVED** [review]

File: scopes/react/ui/loader-fallback/loader-fallback.docs.mdx (hunk 1 above)
Comment (KutnerUri) [reply scopes/react/ui/loader-fallback/loader-fallback.docs.mdx:11]: this is a markdown numbered list

File: scopes/react/ui/loader-fallback/loader-fallback.tsx (hunk 2 above)
Comment (KutnerUri) [reply scopes/react/ui/loader-fallback/loader-fallback.tsx:44]: others should not change, but this supports them if they do

**Pull request merged**